core/network/router.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_ROUTER_HPP
+#define KAGOME_ROUTER_HPP
+
+#include <memory>
+
+namespace kagome::network {
+  /**
+   * Router, which reads and delivers different network messages to the
+   * observers, responsible for their processing
+   * @tparam ChannelType - type of the channel, over which the messages are
+   * read; can be, for example, a Libp2p Stream
+   */
+  template <typename ChannelType>
```

Comment:
I think read/writer abstraction instead of template is fine here

Reply:
ReadWriter belongs to Libp2p, and we don't want to use it in the interfaces